src/main/java/com/google/devtools/build/lib/skyframe/RegisteredExecutionPlatformsFunction.java

```diff
@@ -234,7 +234,10 @@ public boolean shouldRetain(Target target, boolean explicit) {
       if (explicit) {
         return true;
       }
+      return hasPlatformInfo(target);
+    }
 
+    boolean hasPlatformInfo(Target target) {
       // If the rule requires platforms or toolchain resolution, it can't be used as a platform.
```

Comment:
This comment appears misplaced here and indicates that the associated logic maybe doesn't belong in this method?

Reply:
Cleaned up the comment.